substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -379,6 +379,16 @@ pub mod pallet {
 	pub type Nominators<T: Config> =
 		CountedStorageMap<_, Twox64Concat, T::AccountId, Nominations<T>>;
 
+	/// Stakers whose funds are managed by other pallets.
+	///
+	/// This pallet does not apply any locks on them, therefore they are only virtually bonded. They
+	/// are expected to be keyless accounts and hence should not be allowed to mutate their ledger
+	/// directly via this pallet. Instead, these accounts are managed by other pallets and accessed
+	/// via low level apis. We keep track of them to do minimal integrity checks.
+	// TODO(ank4n): Can we keep this entry in `Ledger`? Worth a migration?
+	#[pallet::storage]
+	pub type VirtualStakers<T: Config> = CountedStorageMap<_, Twox64Concat, T::AccountId, ()>;
```

Comment:
Ideally I would like to move this to `StakingLedger` since it is always read in conjunction with that. What we can probably do is, have it like this in the initial version and migrate it as a bool/enum field in ledger along with Currency-Fungible MBM.

cc: @gpestana @kianenigma  thoughts?

Reply:
I would add a new field in the `StakingLedger` and save the extra reads associated with checking if a ledger is virtual if we keep this info in a separate storage. The downside is the migration, but I hope we can get to a place where migrations are not blockers or make us think twice about refactors and design decisions, especially as the MB migrations are used more and more.

Reply:
Because the number of virtual stakers starts at 0 and they are expected to grow over time, but still be a minority compared to regular stakers, I think it's ok to have a separate storage item for this. It would feel bad to migrate all of the ledgers for a few accounts which will become virtual stakers.

Staking functions are storage intensive anyway, but are not called that often and I don't expect this extra read to alter the dynamic too much. Worth mentioning that the ledger logic is already pretty complex, so any addition there has to have a good reason IMO.

If there are a lot of virtual stakers and it starts to make sense to optimize space and compute, we could write a migration in the future.

Reply:
Virtual stakers will be same as number of pools. Though there are like ~180 pools in prod today.

The reason to move it to ledger: 1) It is always read in conjunction with ledger. 2) Ledger impl maintains the lock so we could argue it functionally belongs in Ledger.
Reason not to move: Will require a MBM so we can batch it with Currency-Fungible migration. 

I kind of lean towards not moving since I don't think it is a huge overhead (it does not even try to read the value from storage key). And at some point later we can migrate it into ledger.

Reply:
No strong opinion, but I would also not over optimize thing by adding another migration here. Separate map is fine for now. If and when that extra read is an issue, we will deal with it. Quite possibly it will never be an issue ;)